bindings/wasm/src/did/wasm_document.rs

```diff
@@ -139,20 +238,84 @@ impl WasmDocument {
   ///
   /// Throws an error if the method is not found.
   #[wasm_bindgen(js_name = resolveMethod)]
-  pub fn resolve_method(&self, query: &UDIDUrlQuery) -> Result<WasmVerificationMethod> {
+  pub fn resolve_method(&self, query: &UDIDUrlQuery, scope: Option<WasmMethodScope>) -> Result<WasmVerificationMethod> {
     let method_query: String = query.into_serde().wasm_result()?;
+    if let Some(scope) = scope {
+      return Ok(WasmVerificationMethod(
+        self
+          .0
+          .resolve_method_with_scope(&method_query, scope.0)
+          .ok_or(identity::did::Error::MethodNotFound)
+          .wasm_result()?
+          .clone(),
+      ));
+    }
     Ok(WasmVerificationMethod(
       self.0.try_resolve_method(&method_query).wasm_result()?.clone(),
     ))
```

Comment:
Nit: this can be simplified. Optional.

We should also consider returning an option rather than an error here, or are errors preferred to handle than null in Javascript?

```suggestion
    let method: &IotaVerificationMethod = if let Some(scope) = scope {
      self
        .0
        .resolve_method_with_scope(&method_query, scope.0)
        .ok_or(identity::did::Error::MethodNotFound)
        .wasm_result()?
    } else {
      self.0.try_resolve_method(&method_query).wasm_result()?
    };
    Ok(WasmVerificationMethod(method.clone()))
```